evm_arithmetization/src/generation/prover_input.rs

```diff
@@ -513,103 +517,92 @@ impl<F: RichField> GenerationState<F> {
 
     /// Returns a pointer to a node in the list such that
     /// `node[0] <= addr < next_node[0]` and `addr` is the top of the stack.
-    fn run_next_insert_account(&self) -> Result<U256, ProgramError> {
+    fn run_next_insert_account(&mut self, input_fn: &ProverInputFn) -> Result<U256, ProgramError> {
         let addr = stack_peek(self, 0)?;
-        let accounts_mem = self.memory.get_preinit_memory(Segment::AccountsLinkedList);
-        let accounts_linked_list =
-            LinkedList::<ACCOUNTS_LINKED_LIST_NODE_SIZE>::from_mem_and_segment(
-                &accounts_mem,
-                Segment::AccountsLinkedList,
-            )?;
-
-        if let Some(([.., pred_ptr], [_, ..], _)) =
-            accounts_linked_list
-                .tuple_windows()
-                .find(|&(_, [prev_addr, ..], [next_addr, ..])| {
-                    (prev_addr <= addr || prev_addr == U256::MAX) && addr < next_addr
-                })
-        {
-            Ok(pred_ptr / U256::from(ACCOUNTS_LINKED_LIST_NODE_SIZE))
-        } else {
-            Ok((Segment::AccountsLinkedList as usize).into())
+
+        let (&pred_addr, &pred_ptr) = self
+            .accounts
+            .range(..=addr)
+            .next_back()
+            .unwrap_or((&U256::MAX, &(Segment::AccountsLinkedList as usize)));
```

Comment:
If I'm not misunderstanding here, we'd be having `Segment::AccountsLinkedList` in case it's not found, and at the end returning `Ok(U256::from(pred_ptr / ACCOUNTS_LINKED_LIST_NODE_SIZE))` which would be `Segment::AccountsLinkedList / ACCOUNTS_LINKED_LIST_NODE_SIZE` instead of `Segment::AccountsLinkedList`?

Reply:
But before this PR we were returning `Segment::AccountsLinkedList as usize` if not found, without dividing it.